packages/expo/scripts/autolinking.rb

```diff
@@ -30,4 +30,25 @@ def expo_patch_react_imports!(installer, options = {})
   end
 
   Expo::ReactImportPatcher.new(installer, options).run!
+  expo_patch_for_updates_debug!(installer)
+end
+
+def expo_patch_for_updates_debug!(installer)
+  projects = installer.aggregate_targets
+    .map{ |t| t.user_project }
+    .uniq{ |p| p.path }
+    .push(installer.pods_project)
+
+  ex_updates_native_debug = ENV['EX_UPDATES_NATIVE_DEBUG'] == '1'
+
+  projects.each do |project|
+    project.build_configurations.each do |config|
+      if ex_updates_native_debug
+        config.build_settings['OTHER_CFLAGS'] = "$(inherited) -DEX_UPDATES_NATIVE_DEBUG=1"
+      else
+        config.build_settings['OTHER_CFLAGS'] = "$(inherited)"
```

Comment:
I don't quite understand why `else` and reseting existing flags are needed here

Reply:
Having the else clause allows you to go back to normal debugging by removing the environment variable and doing pod install again.

Reply:
Ah, I see. Check out https://github.com/expo/expo/blob/7aebe42601d6fadd98607fb563499c60bbfdf47a/packages/expo-modules-autolinking/scripts/ios/project_integrator.rb#L105 where we do a similar thing with Swift compiler flags. It removes the flag with `gsub` method.